modules/pubsub/main.tf

```diff
@@ -85,9 +88,19 @@ resource "google_pubsub_topic_iam_member" "pubsub_viewer_role" {
 }
 
 resource "google_pubsub_subscription" "pubsub_subscription" {
-  count   = var.create_subscriber ? 1 : 0
+  count   = (var.create_subscriber && !var.push_subscriber) ? 1 : 0
```

Comment:
I don't think the new variable needs to change the existing behavior at all. Instead, I suggest:

1. If `create_subscriber` is true, create a (pull) subscription (current behaviour)
2. If `create_push_subscriber` is true, create a push subscription (added)

Both can be true, or neither can be true.